src/repo-gulpfile.js

```diff
@@ -153,6 +156,71 @@ const generateVersioningFile = async function () {
     )
 }
 
+
+const generateHomepage = async function () {
+    /* 
+     * If a repository.json is defined at the root of the repository, this function generate a homepage for the repository.
+     * It will be based on repository.json and versioning.json data and homepage.pug file.
+     * 
+     * See homepage.pug for more information on how to create repository.json file
+     */
+
+    //joining path of directory
+    let basePath = process.cwd().substr(0, process.cwd().length - 44)
+    let repositoryDataPath = path.join(basePath, 'repository.json')
+    let versioningFilePath  = path.join(basePath, 'bundles/versioning.json')
+
+    const directoryPath = path.join(basePath, 'src')
+    var promises = []
+
+    // The homepage should only be generated if a repository.json file exist at the root of the repo
+    if (fs.existsSync(repositoryDataPath)) {
+        
+        console.log("Generation of the repository homepage")
+
+        // We need data from repository.json and versioning.json created previously
+        repositoryData = JSON.parse(fs.readFileSync(repositoryDataPath, 'utf8'))
+        extensionsData = JSON.parse(fs.readFileSync(versioningFilePath, 'utf8'))
+
+        // Creation of the list of available extensions
+        // [{name: sourceName, tags[]: []}]
+        extensionList = []
+
+        extensionsData.sources.forEach(extension => {
+            extensionList.push(
+                {
+                    "name": extension.name,
+                    "tags": extension.tags
+                }
+            )
+        })
+
+        repositoryData["sources"] = extensionList
+
+        // repositoryData is then of the form
+        /*
+            {
+                "repositoryName": "",
+                "repositoryDescription": "",
+                "repositoryLogo": "url",
+                "baseURL": "https://yourlinkhere",
+                "sources": [{name: sourceName, tags[]: []}]
+            }
+        */
+
+        // Compilation of the pug file, that should exist in the local directory
+        const htmlCode = pug.compileFile('homepage.pug')(
```

Comment:
```suggestion
                const htmlCode = pug.compileFile(path.join('website-generation', 'homepage.pug'))(
```
I suggest creating a directory called 'website-generation' and putting the pugfile there so that it's not in the root!